### PR TITLE
Review interface

### PR DESCRIPTION
**Pull request by VolkerHartmann**
See comments in logs and code.

### CODE REVIEW

File: allValidatorsWithInterface/src/main/java/edu/kit/scc/dem/first_steps/Main.java [hunk 1]
```diff
@@ -46,7 +46,9 @@ public static void main(String[] args) throws ValidatorInterface.ValidationExcep
 
         try {
             cmd = parser.parse(options, args);
-        } catch (ParseException e) {
+        } catch (ParseException e) {  // Why no tests with invalid input! 
+          // Testing only valid input makes no sense as the handling of invalid input
+          // is much more important.
```
Comment (maximilianiKIT) [inline allValidatorsWithInterface/src/main/java/edu/kit/scc/dem/first_steps/Main.java:60]: You are right. The only problem is that this exception is only thrown when an internal error occurs during the parsing process. This is very difficult to test...

Comment (VolkerHartmann) [reply allValidatorsWithInterface/src/main/java/edu/kit/scc/dem/first_steps/Main.java]: I see. It seems to be a very robust and well tested library. ;-)

Did you also tried a '-' only or '--' or '-longOption'?

Comment (maximilianiKIT) [reply allValidatorsWithInterface/src/main/java/edu/kit/scc/dem/first_steps/Main.java]: Not yet...

Comment (maximilianiKIT) [reply allValidatorsWithInterface/src/main/java/edu/kit/scc/dem/first_steps/Main.java]: It is implemented now...

File: allValidatorsWithInterface/src/main/java/edu/kit/scc/dem/first_steps/validators/ValidatorInterface.java [hunk 2]
```diff
@@ -25,6 +25,20 @@
      */
     boolean isValid(String input) throws ValidationException;
 
+//    // This method should be overwritten.
```
Comment (maximilianiKIT) [inline allValidatorsWithInterface/src/main/java/edu/kit/scc/dem/first_steps/validators/ValidatorInterface.java:29]: Which advantages has this kind of flexibility?

Comment (VolkerHartmann) [reply allValidatorsWithInterface/src/main/java/edu/kit/scc/dem/first_steps/validators/ValidatorInterface.java:29]: Parsing the input is always the same code, right? 
But asking for a "text" to validate is not very helpful. There should be a more precise information. If necessary even more than one line. 
If you are doing so you can simply call the 'askForInputAndValidate()' method. (Ok, for the google library you have to override the method anyway).

**Pull request merged**